collections/_article/dev-snapshot-godot-4-6-dev-6.md

```diff
@@ -0,0 +1,91 @@
+
+---
+
+*The cover illustration is from* [**WAR RATS: The Rat em Up**](https://store.steampowered.com/app/3167260/WAR_RATS_The_Rat_em_Up/?curator_clanid=41324400), *a tactical shoot 'em up, where you lead your fearless troops in the Great Rat War against the onslaught of the TechnoRat legion. You can get the game on [Steam](https://store.steampowered.com/app/3167260/WAR_RATS_The_Rat_em_Up/?curator_clanid=41324400), and check out the developers on [Twitter](https://twitter.com/wootusart) or [Instagram](https://www.instagram.com/wootusart/)!*
+
+## Highlights
+
+### GDScript: Tracy profiler support
+
+In the 4.6 dev 4 blog post, we had a section featuring [dedicated profiler support](/article/dev-snapshot-godot-4-6-dev-4/#buildsystem-support-dedicated-profilers). That section was written for more technical users that were already familiar with how dedicated profilers operated, so it focused on their immediate benefit of profiling the C++ code of the Godot Engine itself. In hindsight, we could've expanded it to be more readable for the average user, as this system had the potential to expand to profiling well beyond our system internals. It was ultimately considered superfluous, as no such integrations existed, and implementing them in a handful of weeks before feature freeze was extremely unlikely.
+
+…You read the header, you know where this is going. [Lukas Tenbrink](https://github.com/Ivorforce), the original implementer of dedicated profiler support, collaborated with [enetheru](https://github.com/enetheru) to bring this native support to GDScript ([GH-113279](https://github.com/godotengine/godot/pull/113279))! While currently limited to tracy, this is a major first step in allowing these dedicated profilers to be a universal solution for debugging performance.
+
+Documentation on how to utilize these native profilers yourself will soon be available [here](https://docs.godotengine.org/en/latest/engine_details/development/debugging/using_cpp_profilers.html) (currently [in PR form](https://github.com/godotengine/godot-docs/pull/11484)).
+
+### Android: Storage Access Framework support
+
+Godot apps on Android have this unfortunate hurdle of effectively requiring the `MANAGE_EXTERNAL_STORAGE` permission. Without it, users were limited to files within the Documents/Download directories that the app itself created. Despite other scoped permissions being available for media files (`READ_MEDIA_IMAGES`, `READ_MEDIA_AUDIO`, and `READ_MEDIA_VIDEO`), non-media files (`.txt`, `.json`, etc) still required the special `MANAGE_EXTERNAL_STORAGE` permission. While workarounds did exist, the best solution would be one that sidesteps the issue entirely.
```

Comment:
I'm not 100% confident in this description, maybe @m4gr3d is around to clarify (@syntaxerror247 is likely sleeping).

This permission is required for apps like the Godot editor which need to be able to edit files outside the app's domain. But far most Godot apps/games don't need external storage access, and this permission is frowned upon as it's inheritently unsafe. The current wording sounds a bit like all Godot apps/games need this permission, which isn't true.